src/librustc_metadata/cstore.rs

```diff
@@ -82,16 +84,22 @@ pub struct CrateMetadata {
     pub dep_kind: Lock<DepKind>,
     pub source: CrateSource,
 
-    pub proc_macros: Option<Vec<FullProcMacro>>,
-
     /// Whether or not this crate should be consider a private dependency
     /// for purposes of the 'exported_private_dependencies' lint
-    pub private_dep: bool
+    pub private_dep: bool,
+
+    pub host_lib: Option<Library>,
+    pub span: Span,
+
+    pub cached_proc_macros: Lock<Option<Lrc<CachedProcMacros>>>
```

Comment:
You don't need this cache, @petrochenkov pointed out elsewhere that the expansion system has its own caching.